src/proposal.ts

```diff
@@ -340,16 +345,18 @@ export class Proposal implements IStateful<IProposalState> {
     )
   }
 
+  public stakingToken() {
+    // TODO: the token to stake with from the proposal, instead ot he generic one
```

Comment:
there are a lot of TODOs in here, have you created issues for all of them to remember?

Reply:
I've addressed some of them;added some issues (both here and in the subgraph) for them; the ones that are left are mostly commens that are more about the code than about fucntionality ("write this test"), and I've just left them there..